syntax = "proto3";

package meterusage;

service MeterUsage {
  // RPC for retrieving meter usage data

  // Since there are numerous meter usage data points collected over time, it's better to stream the results
  // rather sending a single enormous response payload.
  rpc ReadData(ReadRequest) returns (stream MeterData) {}
}

// The ReadRequest message can be extended to accept filters such as:
// timestamp_from
// timestamp_to
message ReadRequest {}

message MeterData {
  // Date and time of recording, stored as a Unix timestamp.
  int64 timestamp = 1;

  // Meter usage value.
  // Using double instead of float due to an issue with Python losing decimal precision during gRPC transmission
  double value = 2;
}